internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go

```diff
@@ -607,3 +603,20 @@ func generateNewNumaflowControllerDef(nfcRollout *apiv1.NumaflowControllerRollou
 
 	return newNumaflowControllerDef, nil
 }
+
+// areDependentResourcesDeleted checks if dependent resources are deleted.
```

Comment:
```suggestion
// areDependentResourcesDeleted checks if dependent resources are deleted.
// note we only look for the resources whose Numaflow children have finalizers 
// (i.e. pipeline, isbsvc, not monovertex) since these are the ones for which Numaflow 
// Controller must be running in order to remove the finalizer
```